[PATCH] From 92ab2d74fca06f86c00d886ac249f7f2d89e93fe Mon Sep 17

From: Akhilesh Samineni <akhilesh.samineni@broadcom.com>
Date: Mon, 5 Apr 2021 13:21:40 -0700
 00:00:00 2001 Subject: [PATCH 8/8] Link local scope was not set while binding
 socket  with local address causing socket errors for bgp ipv6 link local
 neighbors.
---
 bgpd/bgp_network.c |    5 +++++
 1 file changed, 5 insertions(+)

diff --git a/bgpd/bgp_network.c b/bgpd/bgp_network.c
index d535c3d87..c9b019424 100644
--- a/bgpd/bgp_network.c
+++ b/bgpd/bgp_network.c
@@ -753,6 +753,11 @@ int bgp_update_address(struct interface *ifp, const union sockunion *dst,
 		return 1;
 
 	prefix2sockunion(sel, addr);
+
+	if (IN6_IS_ADDR_LINKLOCAL(&addr->sin6.sin6_addr)) {
+		addr->sin6.sin6_scope_id = ifp->ifindex;
+	}
+
 	return 0;
 }
 
